skorch/helper.py

```diff
@@ -508,3 +509,83 @@ def describe_signature(self, df):
             )
 
         return signature
+
+
+class AccelerateMixin:
+    """Mixin class to add support for huggingface's accelerate
+
+    *Experimental*
+
+    Examples
+    --------
+    >>> from skorch import NeuralNetClassifier
+    >>> from skorch.helper import AccelerateMixin
+    >>> from accelerate import Accelerator
+
+    >>> class AcceleratedNet(AccelerateMixin, NeuralNetClassifier):
+    >>>     pass
+
+    >>> accelerator = Accelerator(...)
+    >>> net = AcceleratedNet(
+    ...     MyModule,
+    ...     accelerator=accelerator,
+    ...     device=None,
+    ...     callbacks__print_log__sink=accelerator.print)
+    >>> net.fit(X, y)
+
+    """
+    def __init__(self, *args, accelerator, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.accelerator = accelerator
+
+        if self.accelerator.device_placement and (self.device is not None):
+            raise ValueError(
+                "When device placement is performed be accelerator, set device=None")
+
+    def _initialize_criterion(self, *args, **kwargs):
+        super()._initialize_criterion(*args, **kwargs)
+
+        with self._current_init_context('criterion'):
+            for name in self._criteria:
+                criterion = getattr(self, name + '_')
+                if isinstance(criterion, torch.nn.Module):
+                    self.accelerator.prepare(criterion)
+
+            return self
+
+    def _initialize_module(self, *args, **kwargs):
+        super()._initialize_module(*args, **kwargs)
+
+        with self._current_init_context('module'):
+            for name in self._modules:
+                module = getattr(self, name + '_')
```

Comment:
The pattern of `getattr(self, ... + '_')` is used quite a bit in the codebase. Makes me think storing `_` may have been a bit nicer.

(**Not actionable here**)

Reply:
I agree. I was thinking about something like:

```python
        def get_modules(self):
            for name in self._modules:
                module = getattr(self, name + '_')
                if isinstance(module, nn.module):
                    yield module
```

and replacing all those lines like the quoted one with a call to that method. Same for criteria and optimizers. But that would be for another PR.